nucliadb/nucliadb/train/generators/image_classifier.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+import json
+from typing import Any, AsyncIterator, Dict, List, Tuple, cast
+
+from nucliadb_protos.dataset_pb2 import (
+    ImageClassification,
+    ImageClassificationBatch,
+    TrainSet,
+)
+from nucliadb_protos.nodereader_pb2 import StreamRequest
+from nucliadb_protos.resources_pb2 import FieldType, PageStructure, VisualSelection
+
+from nucliadb.common.cluster.abc import AbstractIndexNode
+from nucliadb.ingest.fields.base import Field
+from nucliadb.ingest.fields.file import File as FileField
+from nucliadb.ingest.fields.link import Link as LinkField
+from nucliadb.ingest.orm.resource import KB_REVERSE, Resource
+from nucliadb.train import logger
+from nucliadb.train.generators.utils import get_resource_from_cache
+
+VISUALLY_ANNOTABLE_FIELDS = {FieldType.FILE, FieldType.LINK}
+
+# PAWLS JSON format
+PawlsPayload = Dict[str, Any]
+
+
+async def generate_image_classification_payloads(
+    kbid: str,
+    trainset: TrainSet,
+    node: AbstractIndexNode,
+    shard_replica_id: str,
+) -> AsyncIterator[ImageClassificationBatch]:
+    request = StreamRequest()
+    request.shard_id.id = shard_replica_id
+    request.reload = True
+    batch = ImageClassificationBatch()
+    async for item in node.stream_get_fields(request):
+        rid = item.uuid
+        resource = await get_resource_from_cache(kbid, rid)
```

Comment:
NIT: Does this mean that if the resource is not in the cache it will return None?
If not, the naming of this util could be improved to something more accurate like: `get_resource_from_cache_or_db` or `cached_get_resource`?